contracts/Company/Extras/ExitController.sol

```diff
@@ -0,0 +1,281 @@
+    // Immutable state
+    ////////////////////////
+
+    // a root of trust contract
+    Universe private UNIVERSE;
+    IERC223Token private EURO_TOKEN;
+    // equity token from ETO
+    IEquityToken private EQUITY_TOKEN;
+
+    ////////////////////////
+    // Mutable state
+    ////////////////////////
+
+    // controller lifecycle state
+    State private _state;
+
+    // exit values get set when exit proceedings start
+    uint256 private _exitEquityTokenSupply = 0;
+    uint256 private _exitAquisitionPriceEurUlps = 0;
+    uint256 private _manualPayoutResolutionStart = 0;
+
+    // keep record of manually resolved payout
+    mapping(address => bool) private payoutManuallyResolved;
+
+    ////////////////////////
+    // Modifiers
+    ////////////////////////
+
+    ////////////////////////
+    // Constructor
+    ////////////////////////
+
+    constructor(
+        Universe universe,
+        IEquityToken equityToken
+    )
+        public
+        Agreement(universe.accessPolicy(), universe.forkArbiter())
+        Reclaimable()
+    {
+        UNIVERSE = universe;
+        EURO_TOKEN = UNIVERSE.euroToken();
+        EQUITY_TOKEN = equityToken;
+        _state = State.Setup;
+    }
+
+    ////////////////////////
+    // External functions
+    ////////////////////////
+
+    //
+    // Implements IControllerGovernance
+    //
+    function state()
+        public
+        constant
+        returns (State)
+    {
+        return _state;
+    }
+
+    function payoutInfo()
+        public
+        constant
+        returns (uint256 exitEquityTokenSupply, uint256 exitAquisitionPriceEurUlps, uint256 manualPayoutResolutionStart)
+    {
+        return (
+            _exitEquityTokenSupply, _exitAquisitionPriceEurUlps, _manualPayoutResolutionStart
+        );
+    }
+
+    // calculate how many eurotokens one would receive for the given amount of tokens
+    function eligibleProceedsForTokens(uint256 amountTokens)
+        public
+        constant
+        returns (uint256)
+    {
+        if (_state == State.Setup ) {
+            return 0;
+        }
+        // calculate the amount of eligible proceeds based on the total equity token supply and the
+        // acquisition price
+        return Math.mul(_exitAquisitionPriceEurUlps, amountTokens) / _exitEquityTokenSupply;
+    }
+
+    // calculate how many eurotokens the user with the given address would receive
+    function eligibleProceedsForInvestor(address investor)
+        public
+        constant
+        returns (uint256 equityTokens, uint256 proceeds)
+    {
+        equityTokens = 0;
+        proceeds = 0;
+
+        if (payoutManuallyResolved[investor]) {
+            return;
+        }
+
+        if (_state == State.Payout) {
+            equityTokens = EQUITY_TOKEN.balanceOf(investor);
+        }
+        else if (_state == State.ManualPayoutResolution) {
+            equityTokens = EQUITY_TOKEN.balanceOfAt(investor, _manualPayoutResolutionStart);
```

Comment:
We can add support for users without claimed tokens here by interacting with the `ETOCommitment` smart-contract.
Pseudo-code 
```
  if (equityTokens == 0) {
                (claimedOrRefunded, equityTOkenAmount) = ETO_COMMITMENT.investorTicket(investor)

                if (claimedOrRefund == false) {
                    equityTokens = equityTOkenAmount;
                }
            }
           
```